google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelector.java

```diff
@@ -550,14 +553,20 @@ protected void doLogin() {
     }
   }
 
-  private void onSelectionChanged(ResourceProjectModelItem newSelection) {
+  private void onSelectionChanged(GcpModelItem newSelection) {
     CredentialedUser user = null;
     ProjectSelectionChangedEvent event = null;
     if (newSelection != null) {
       if (newSelection.getParent() instanceof GoogleUserModelItem) {
         user = ((GoogleUserModelItem) newSelection.getParent()).getCredentialedUser();
       }
-      event = new ProjectSelectionChangedEvent(newSelection.getProject(), user);
+
+      if (newSelection instanceof InvalidResourceProjectModelItem) {
+        event = new ProjectSelectionChangedEvent(null, user);
```

Comment:
This doesn't look safe. For example, `AppEngineApplicationPanel#refresh` listens on this event and doesn't account for null `Project`s.

Reply:
There is a check in AppEngineApplicationInfoPanel#refresh for null project

Reply:
Ah ok I missed that you updated it.

Reply:
On second look, could you just pass in the `InvalidResourceProjectModelItem` here instead of null? That way you could make the paramenter not nullable and then do the check for the type inside of the event itself. Also allowing you to get rid of the if/else here.

Reply:
I won't be able to pass the  InvalidResourceProjectModelItem here because the signature is ProjectSelectionChangedEvent(Project, CredentialedUser)

Reply:
ah right - sgtm